data_store/table_extraction/process.py

```diff
@@ -75,38 +76,45 @@ def process(self, table: Table) -> None:
         :param table: The table to process.
         :return: None
         """
+
         self._strip_whitespace(table)
         self._lift_header(table)
         self._remove_merged_headers(table)
         self._drop_cols_by_name(table)
         self._remove_ignored_non_header_rows(table)
         self._replace_dropdown_placeholder(table)
         self._drop_bespoke_rows(table)
+
         if self.drop_empty_rows:
             self._drop_empty_rows(table)
+
         if self.drop_empty_tables and table.df.empty:
-            table.df = None
+            table.df = pd.DataFrame()
 
     def _lift_header(self, table: Table) -> None:
-        header = table.df.iloc[: self.num_header_rows]
-        header = self._concatenate_headers(header, headers_to_ffill=self.merged_header_rows)
-        table.df.columns = header
+        header: DataFrame = table.df.iloc[: self.num_header_rows]
+        table.df.columns = Index(
+            self._concatenate_headers(
+                header,
+                headers_to_ffill=self.merged_header_rows,
+            )
+        )
         table.df = table.df.iloc[self.num_header_rows :]
 
     @staticmethod
     def _remove_merged_headers(table: Table) -> None:
-        prev_col = ("", "")
+        prev_col: tuple[str, int] = ("", -1)  # Initialize with valid types
```

Comment:
This is a but hard to decipher because some seem to like using walrus operators in comprehension lists.
Long story short `prev_col` stores `(col_name: str, idx: int)` but is initialised incorrectly as `(str, str)`.

Reply:
```suggestion
        prev_col: tuple[str, int]
```

It is, I believe, possible to do this as well if the initialisation is not actually useful